### PR TITLE
CLDIDE 2682 : Add startPoint parameter into project source storage and change git import behavior

### PR DESCRIPTION
**Pull request by akorneta**

### CODE REVIEW

**Comment by azatsarynnyy** [comment]
@akorneta are you sure that **GitCheckoutAction** should be in **che-core-ide-app**?

**Comment by akorneta** [comment]
@azatsarynnyy my fault it should be moved to git-plugin

**Comment by azatsarynnyy** [comment]
sure :smile:

File: platform-api-client-gwt/che-core-client-gwt-git/src/main/java/org/eclipse/che/api/git/gwt/client/GitServiceClient.java [hunk 1]
```diff
@@ -113,6 +114,21 @@ void branchList(String workspaceId,
                     AsyncRequestCallback<List<Branch>> callback);
 
     /**
+     * Get the list of the branches. For now, all branches cannot be returned at once, so the parameter <code>remote</code> tells to get
```
Comment (azatsarynnyy) [inline platform-api-client-gwt/che-core-client-gwt-git/src/main/java/org/eclipse/che/api/git/gwt/client/GitServiceClient.java]: Please, review these javadocs.
Seems it was copypasted from somewhere :smile:

**Comment by vparfonov** [comment]
ok, after release 4.0-beta13

**Comment by skabashnyuk** [comment]
ok

File: platform-api/che-core-api-git/src/main/java/org/eclipse/che/api/git/GitProjectImporter.java [hunk 2]
```diff
@@ -132,11 +136,12 @@ public void importSources(FolderEntry baseFolder,
             final String localPath = localPathResolver.resolve((VirtualFileImpl)baseFolder.getVirtualFile());
             final DtoFactory dtoFactory = DtoFactory.getInstance();
             final String location = storage.getLocation();
+            final String repositoryName = location.substring(location.lastIndexOf("/") + 1, location.length() - 4);
```
Comment (sleshchenko) [inline platform-api/che-core-api-git/src/main/java/org/eclipse/che/api/git/GitProjectImporter.java]: Please add comment with description and example. Why "/"+1 and location.length() - 4...

**Comment by sleshchenko** [comment]
OK for me

File: platform-api/che-core-api-git/src/main/java/org/eclipse/che/api/git/GitProjectImporter.java (hunk 2 above)
Comment (akorneta) [inline platform-api/che-core-api-git/src/main/java/org/eclipse/che/api/git/GitProjectImporter.java]: ok

**Comment by skabashnyuk** [comment]
ok

**Pull request merged**